services/supervisord/logs.go

```diff
@@ -269,3 +289,157 @@ func readURL(ctx context.Context, url string) ([]byte, error) {
 	}
 	return b, nil
 }
+
+func addAdminSummary(ctx context.Context, archive *zip.Writer) error {
+	tmpfile, err := ioutil.TempFile("", "*-pmm-admin-summary.zip")
+	if err != nil {
+		return err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return err
+	}
+
+	adminArgs := [][]string{
+		{"summary", "--filename", tmpfile.Name(), "--skip-server"},
+		{"summary", "--filename", tmpfile.Name()},
+	}
+
+	// Try with and without --skip-server just in case the installed pmm-admin version doesn't support it
+	for _, args := range adminArgs {
+		cmd := exec.CommandContext(ctx, "pmm-admin", args...) // nolint
+		pdeathsig.Set(cmd, unix.SIGKILL)
+		var out []byte
+		out, err = cmd.CombinedOutput()
+		log.Println(string(out))
+
+		if err != nil {
+			continue
+		}
+		break
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "cannot run pmm-admin summary")
+	}
+
+	tmpDir, err := ioutil.TempDir("", "pmm-admin-summary")
+	if err != nil {
+		return err
+	}
+
+	if err := unzip(tmpfile.Name(), tmpDir); err != nil {
+		return err
+	}
+
+	clientDir := filepath.Join(tmpDir, "client")
+	if _, err := os.Stat(clientDir); err != nil {
+		return err
+	}
+
+	if err := addToZip(clientDir, archive); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func unzip(archive, target string) error {
```

Comment:
There is no need to unzip a file into directory – we should be able to manipulate zip file headers completely in memory.